spec/acceptance/instance_spec.rb

```diff
@@ -231,4 +232,150 @@ def expect_failed_apply(config)
       expect(instance.state.name).to eq('running')
     end
   end
+
+  describe 'should create a new instance with puppet resource' do
+
+    before(:all) do
+      @config = {
+        :name => "#{PuppetManifest.env_id}-#{SecureRandom.uuid}",
+        :instance_type => 't1.micro',
+        :region => @default_region,
+        :image_id => 'ami-41e85d5c',
+        :ensure => 'present',
+        :monitoring => false,
+        :availability_zone => @default_availability_zone,
+        :security_groups => 'default',
+      }
+
+      # The value for this ENV var must be an existing key in your Amazon account
+      # the key must be available in the sa-east-1 region
+      @config[:key_name] = ENV['AWS_KEY_PAIR'] if ENV['AWS_KEY_PAIR']
+      # create new instance with puppet resource
+      TestExecutor.puppet_resource('ec2_instance', @config, '--modulepath ../')
+      #wait for instance to report as running
+      @instance = get_instance(@config[:name])
+      @aws.ec2_client.wait_until(:instance_running, instance_ids:[@instance.instance_id])
+      # set env variable and use puppet resource to inspect state of ec2 instance
+      ENV['AWS_REGION'] = @config[:region]
+      @result = TestExecutor.puppet_resource('ec2_instance', {:name => @config[:name]}, '--modulepath ../')
+      expect(@result.stderr).not_to match(/\b/)
+      # re-assign @instance for more up to date info
+      @instance = get_instance(@config[:name])
+    end
+
+    after(:all) do
+      @config[:ensure] = 'absent'
+      TestExecutor.puppet_resource('ec2_instance', @config, '--modulepath ../')
+      @aws.ec2_client.wait_until(:instance_terminated, instance_ids:[@instance.instance_id]) do |w|
+        w.max_attempts = 5
+        # this will fail until CLOUD-202 is resolved
+      end
+    end
+
+    it 'ensure is correct' do
+      regex = /(ensure)(\s*)(=>)(\s*)('present')/
+      expect(@result.stdout).to match(regex)
+    end
+
+    it 'instance_type is correct' do
+      regex = /(instance_type)(\s*)(=>)(\s*)('#{@config[:instance_type]}')/
+      expect(@result.stdout).to match(regex)
+    end
+
+    it 'region is correct' do
+      regex = /(region)(\s*)(=>)(\s*)('#{@config[:region]}')/
+      expect(@result.stdout).to match(regex)
+    end
+
+    it 'image_id is correct' do
+      regex = /(image_id)(\s*)(=>)(\s*)('#{@config[:image_id]}')/
+      expect(@result.stdout).to match(regex)
+    end
+
+    it 'security_groups is correct' do
+      pending('this test is blocked by CLOUD-196')
```

Comment:
@zreichert you can unblock this test and update this PR now that #71 is merged.

Thanks!